pkg/flowkit/services/scripts.go

```diff
@@ -52,6 +53,35 @@ func NewScripts(
 
 // Execute script code with passed arguments on the selected network.
 func (s *Scripts) Execute(script *flowkit.Script, network string) (cadence.Value, error) {
```

Comment:
I think a better approach would be to define a `ScriptQuery` type like so:
```go
type ScriptQuery struct {
    ID flow.Identifier
    Height uint64
}
```

And then you can just add that:
```
Execute(script *flowkit.Script, network string, *ScriptType)
```
And maybe even better yet to then also support that on gateways.

Reply:
Good to know! I will proceed with implementing this approach then :pray:

Reply:
> And maybe even better yet to then also support that on gateways.

That actually causes an import cycle error

```bash
imports github.com/onflow/flow-cli/pkg/flowkit/gateway: import cycle not allowed
```

Because the `ScriptQuery` type is defined on the `*services.Scripts` type, where we also import the `Gateway` interface, and the interface itself improves the latter:

```go
ExecuteScript([]byte, []cadence.Value, *services.ScriptQuery) (cadence.Value, error)
```

Do you have any way around this?

Reply:
Pushed the changes so far, https://github.com/onflow/flow-cli/pull/894/commits/c56281b73442a91e4b5f1d876788ece643acc86e. A lot of code was removed, so I like the result so far, from your approach :rocket:

Reply:
Yeah, Gateway shouldn't depend on the types from scripts. I suggest putting ScriptQuery in the flowkit package for now, that's the place where everything will be after the changes I'm working on.

Reply:
Fixed in https://github.com/onflow/flow-cli/pull/894/commits/5c30a8ef508e81ed15c406874f41b66b0739e7d5